### ISSUE TITLE
Add analytics for which states/categories/stories are viewed

### ISSUE THREAD

**Issue by aappling-usgs** [USGS-VIZLAB/water-use-15#15]

**Comment by lindsaycarr** [comment]
Need to wait for interface (buttons, etc) to be a little more finalized before starting this task.

**Comment by wdwatkins** [comment]
Is this ready to go yet?

**Comment by aappling-usgs** [comment]
I think so! Would be great if you could take it on.

**Comment by ldecicco-USGS** [comment]
Reminder to me (and maybe others):
https://developers.google.com/analytics/devguides/collection/gtagjs/migration
That's what we've moved to.

**Comment by ldecicco-USGS** [comment]
Wondering if we'll want to (and it's reasonable to?) track the hovers over the categories. Personally, I only hover, so only recording clicks on the categories might not tell us much.

Other than that, we want to see when people went from national -> (which) state, and (which) state -> (which) state, and (which) state -> national

Once we see the stories we can think about the other stuff.

**Comment by wdwatkins** [comment]
we tracked state hovers for the first water use viz

**Comment by wdwatkins** [comment]
Hmm, I can't speak to performance impacts, but I wouldn't think it matters?  Googling it seems inconclusive, but everything should be loaded by that point, so I wouldn't think there is anything to hold up?

**Comment by jread-usgs** [comment]
Agreed that this was useful info last time, but also note that David made some special code so that it didn't end up triggering too many events for the hovers.

**Comment by aappling-usgs** [comment]
@wdwatkins what is left to do on this issue?

**Comment by wdwatkins** [comment]
We could add scroll depth tracking for key stories, I've not implemented that before though.  There are a few tweaks that could be made to what I submitted in my PR also (mainly the namespacing, but we could also record the previous category in the category change event.

**Comment by aappling-usgs** [comment]
* Scroll depth tracking for key stories sounds like a great idea. I think we've done that in previous vizzies - hopefully there's someone around who remembers.
* Recording the previous category change seems like a pretty easy change, right? If so, I'm in favor.
* I added issue #127 to deal with namespacing for all sorts of global variables all at once. Let's keep that one as a separate issue from this one.

**Comment by aappling-usgs** [comment]
I've opened #146 to cover scroll depth tracking. Recording the previous category is getting done in #137 (https://github.com/USGS-VIZLAB/water-use-15/pull/137/files#diff-2afd72a00f31d3d53312dce9ff03a875R177). So I think we're good for basic analytics and can follow up with #127 and #147 to refine.

@wdwatkins , do you have an estimate of total hours spent on this issue to date?

**Comment by wdwatkins** [comment]
5?  Kinda got entangled with other issues.

**Comment by aappling-usgs** [comment]
Yeah, I hear ya. ~5 is just fine.
